pkg/agents/team4/finalAgent/trustScoreUpdates.go

```diff
@@ -0,0 +1,111 @@
+	reqMsg, err := a.typeAssertRequestMessage(sentMsg)
+	if err != nil {
+		log.Error(err)
+	} else {
+		if food.FoodType(reqMsg.Request()) <= a.CurrPlatFood() {
+			a.AddToGlobalTrust(a.params.coefficients[1])
+			a.Log("Team4: For Requested Food to Leave less than or equal to Food on platform", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.CurrPlatFood(), "global_trust": a.params.globalTrust})
+		} else if food.FoodType(reqMsg.Request()) > a.CurrPlatFood() {
+			a.AddToGlobalTrust(-a.params.coefficients[1])
+			a.Log("Team4: For Requested Food to Leave greater than Food on platform", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.CurrPlatFood(), "global_trust": a.params.globalTrust})
+		}
+	}
+}
+
+func (a *CustomAgentEvo) updateGlobalTrustforReqTakeFood(msg messages.BoolResponseMessage, sentMsg messages.Message) {
+	a.Log("Team4 reponse message received", infra.Fields{"sentMsg_Type": sentMsg.MessageType(), "senderFloor": msg.SenderFloor()})
+	reqMsg, err := a.typeAssertRequestMessage(sentMsg)
+	if err != nil {
+		log.Error(err)
+	} else {
+		if food.FoodType(reqMsg.Request()) >= a.NeighbourFoodEaten() {
+			a.Log("Team4: For Requested Food to Take greater then or equal neighbour food eaten", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.NeighbourFoodEaten(), "global_trust": a.params.globalTrust})
+			a.AddToGlobalTrust(a.params.coefficients[1])
+		} else if food.FoodType(reqMsg.Request()) < a.NeighbourFoodEaten() {
+			a.Log("Team4: For Requested Food to Take less than neighbour food eaten", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.NeighbourFoodEaten(), "global_trust": a.params.globalTrust})
+			a.AddToGlobalTrust(-a.params.coefficients[1])
+		}
+	}
+}
+
+func (a *CustomAgentEvo) CheckForResponse(msg messages.BoolResponseMessage) {
+	if len(a.params.responseMessages) > 0 {
+		for i, respMsg := range a.params.responseMessages { // Iterate through each response message
+			resMsg, err := a.typeAssertResponseMessage(respMsg)
+			if err != nil {
+				log.Error(err)
+			} else {
+				for j, sentMsg := range a.params.sentMessages { // Iterate through each sent message
+					if resMsg.RequestID() == sentMsg.ID() {
+						if a.PlatformOnFloor() && sentMsg.MessageType() == messages.RequestLeaveFood && a.Floor()-msg.SenderFloor() == 1 { // Check if there are any responses messages.
+							a.updateGlobalTrustReqLeaveFood(msg, sentMsg)
+
+						} else if a.params.lastFoodTaken+a.CurrPlatFood() != a.params.lastPlatFood && sentMsg.MessageType() == messages.RequestTakeFood && a.Floor()-msg.SenderFloor() == -1 {
+							a.updateGlobalTrustforReqTakeFood(msg, sentMsg)
+						}
+						a.params.sentMessages = remove(a.params.sentMessages, j) // Remove the accessed response/sent messages from memory
+						a.params.responseMessages = remove(a.params.responseMessages, i)
+						break
+					}
+				}
+			}
+		}
```

Comment:
I would make this a function.

You can also just return here then you don't need the else.